torch/csrc/distributed/autograd/context/dist_autograd_container.cpp

```diff
@@ -5,56 +5,94 @@ namespace torch {
 namespace distributed {
 namespace autograd {
 
-constexpr int kContextIdBits = 48;
-constexpr int64_t kContextIdMask = (1LL << kContextIdBits) - 1;
+constexpr int kAutoIncrementBits = 48;
+constexpr int64_t kAutoIncrementMask = (1LL << kAutoIncrementBits) - 1;
 constexpr int kMaxWorkerId = 65535;
-constexpr int64_t kMaxContextId = kContextIdMask;
 
-thread_local int64_t DistAutogradContainer::current_context_id_ = -1;
+// Each thread has a single autograd_context_id valid at any point in time.
+static thread_local int64_t current_context_id_ = -1;
 
 DistAutogradContainer::DistAutogradContainer()
-    : next_context_id_(0), worker_id_(0), initialized_(false) {}
+    : next_context_id_(0),
+      worker_id_(0),
+      initialized_(false),
+      next_autograd_message_id_(0),
+      max_id_(0) {}
 
 DistAutogradContainer& DistAutogradContainer::init(int64_t worker_id) {
   TORCH_CHECK(
       worker_id >= 0 && worker_id <= kMaxWorkerId,
       "worker_id needs to be in the range [0, 65535]")
 
-  auto& container = getInstance();
+  auto& container = getInstanceInternal();
+  TORCH_CHECK(
+      !container.initialized_,
+      "Container is already initialized! Cannot initialize it twice!");
```

Comment:
Is initialization idempotent? I.e., is it safe to just ignore subsequent initializations? What kind of bug are we catching here by throwing an error?